examples/network_traffic_prediction/nr_models.py

```diff
@@ -0,0 +1,702 @@
+
+        # get all demands features
+        inventory_nfeats = g.nodes[infer_ntype].data['inventory_amounts']
+        th_inventory_nfeats = inventory_nfeats[th.arange(inventory_nfeats.shape[0])]
+
+        predict_results = None
+        for predict_step in range(init_step, (init_step + infer_steps)):
+            res = node_mini_batch_gnn_predict(self._model, loader, return_proba,
+                                                return_label=False, use_ar=True,
+                                                predict_step=predict_step)
+            preds = res[0]
+
+            replace_step = predict_step + window_size
+            # Replace the next step inventory_amounts with the predicitons
+            th_inventory_nfeats[:, replace_step] = preds[infer_ntype].squeeze()
+            inventory_nfeats[th.arange(th_inventory_nfeats.shape[0])] = th_inventory_nfeats
+            g.nodes[infer_ntype].data['inventory_amounts'] = inventory_nfeats
+
+            if predict_results is None:
+                predict_results = preds[infer_ntype]
+            else:
+                predict_results = th.concat([predict_results, preds[infer_ntype]], dim=-1)
+
+        total_predict_results = {infer_ntype: predict_results}
+
+        nid_shuffler = None
+        g = loader.data.g
+
+        if save_prediction_path is not None:
+            # save_embed_path may be None. In that case, we need to init nid_shuffler
+            if nid_shuffler is None:
+                nid_shuffler = NodeIDShuffler(g, node_id_mapping_file,
+                                              list(total_predict_results.keys())) \
+                    if node_id_mapping_file else None
+            shuffled_preds = {}
+            for ntype, pred in total_predict_results.items():
+                pred_nids = loader.target_nidx[ntype]
+                if node_id_mapping_file is not None:
+                    pred_nids = nid_shuffler.shuffle_nids(ntype, pred_nids)
+                shuffled_preds[ntype] = (pred, pred_nids)
+
+            save_node_prediction_results(shuffled_preds, save_prediction_path)
+
+        barrier()
+
+
+def node_mini_batch_gnn_predict(model, loader, return_proba=True,
+                                return_label=False, use_ar=False, predict_step=-1):
+    device = model.device
+    data = loader.data
+    g = data.g
+    preds = {}
+    # For this customized model, we only use one target node type
+    target_ntype = list(loader.target_nidx.keys())[0]
+
+    if return_label:
+        assert loader.label_field is not None, \
+            "Return label is required, but the label field is not provided when" \
+            "initlaizing the loader."
+
+    embs = {}
+    labels = {}
+    model.eval()
+
+    len_dataloader = max_num_batch = len(loader)
+    tensor = th.tensor([len_dataloader], device=device)
+    if gs.utils.is_distributed():
+        th.distributed.all_reduce(tensor, op=th.distributed.ReduceOp.MAX)
+        max_num_batch = tensor[0]
```

Comment:
Are you handling the case when some inferers have less num_batches than others?

Reply:
No. This code was copied from https://github.com/awslabs/graphstorm/blob/66b19983b34f82cd5b79c4fad73b5b6e08d8e0fd/python/graphstorm/model/node_gnn.py#L235.

And no change made to the for loop.